### PR TITLE
Task #13 Implement NPath Metric

### PR DESCRIPTION
**Pull request by zuoqin**

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat elect

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#16](https://github.com/yegor256/aibolit/issues/16); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by g4s8** [comment]
@0crat assign @paulodamaso

**Comment by 0crat** [comment]
> @0crat assign @paulodamaso ([here](https://github.com/yegor256/aibolit/issues/16#issuecomment-582357979))

@g4s8 The job [#16](https://github.com/yegor256/aibolit/issues/16) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/1269ab1b-133b-409e-be7d-523e88e6c093); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: test/metrics/npath/test_all_types.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import unittest
+
+
+class JavaTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super(JavaTestCase, cls).setUpClass()
+
+    def runAnalysis(self):
```
Comment (paulodamaso) [inline test/metrics/npath/test_all_types.py]: @zuoqin PLease split each test case in an isolated test method

File: test/metrics/npath/ooo.java [hunk 2]
```diff
@@ -0,0 +1 @@
+jkhhhk
```
[the remark this answers is not in the document]
Comment (zuoqin) [reply test/metrics/npath/ooo.java]: This file used for UnitTest incorrectly formatted Java file, in this case UnitTest should catch Exception

File: aibolit/metrics/npath/main.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            else:
+                raise Exception('File ' + self.input + ' does not exist')
+            tmppath = os.path.dirname(os.path.realpath(__file__))
+            shutil.copyfile(os.path.join(tmppath, 'pom.xml'), root + '/pom.xml')
+            shutil.copyfile(os.path.join(tmppath, 'npath.xml'), root + '/npath.xml')
+            if showoutput:
+                subprocess.run(['mvn', 'pmd:pmd'], cwd=root)
+            else:
+                subprocess.run(['mvn', 'pmd:pmd'], cwd=root,
+                               stdout=subprocess.DEVNULL,
```
[the remark this answers is not in the document]
Comment (zuoqin) [reply aibolit/metrics/npath/main.py]: @paulodamaso, I double ckecked with flake8 and the general rule is: next line with function arguments starts immediately after ( bracket opens on the previous line

**Comment by paulodamaso** [comment]
@yegor256 This PR is good to merge

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Pull request merged**